pkg/services/ngalert/notifier/alertmanager.go

```diff
@@ -310,22 +305,18 @@ func (am *Alertmanager) applyAndMarkConfig(ctx context.Context, hash string, cfg
 	return nil
 }
 
-func (am *Alertmanager) WorkingDirPath() string {
-	return filepath.Join(am.Settings.DataPath, workingDir, strconv.Itoa(int(am.orgID)))
+func (am *Alertmanager) AppURL() string {
+	return am.Settings.AppURL
 }
 
 // buildIntegrationsMap builds a map of name to the list of Grafana integration notifiers off of a list of receiver config.
-func (am *Alertmanager) buildIntegrationsMap(receivers []*alertingNotify.APIReceiver, templates *alertingTemplates.Template) (map[string][]*alertingNotify.Integration, error) {
-	integrationsMap := make(map[string][]*alertingNotify.Integration, len(receivers))
+func (am *Alertmanager) buildIntegrationsMap(receivers []*alertingNotify.APIReceiver) map[string]*alertingNotify.APIReceiver {
+	integrationsMap := make(map[string]*alertingNotify.APIReceiver, len(receivers))
 	for _, receiver := range receivers {
-		integrations, err := am.buildReceiverIntegrations(receiver, templates)
-		if err != nil {
-			return nil, err
-		}
-		integrationsMap[receiver.Name] = integrations
+		integrationsMap[receiver.Name] = receiver
 	}
 
-	return integrationsMap, nil
+	return integrationsMap
```

Comment:
I think we can remove it at all if we make AlertmanagerConfig.Receviers be a list, which how it is  represented in upstream Alertmanager config